src/games/strategy/triplea/ai/AdvancedUtils.java

```diff
@@ -86,7 +86,7 @@ public static Route trimRouteBeforeFirstTerWithEnemyUnits(final Route route, fin
   public static List<Territory> getTerritoriesWithinXDistanceOfYMatchingZ(final GameData data, final Territory start,
       final int maxDistance, final Match<Territory> match) {
     return getTerritoriesWithinXDistanceOfYMatchingZAndHavingRouteMatchingA(data, start, maxDistance, match,
-        Match.ALWAYS_MATCH);
+        (Match<Territory>)Match.ALWAYS_MATCH);
```

Comment:
Yeah, I don't really like this change as it adds a bunch of ugly casts. I think we should consider solving this a different way as you suggest.